test/integration/smoke/test_deploy_vms_in_parallel.py

```diff
@@ -0,0 +1,173 @@
+                             VirtualMachine)
+from marvin.lib.common import (get_zone,
+                               get_domain,
+                               get_test_template)
+from marvin.cloudstackAPI import (deployVirtualMachine,
+                                  queryAsyncJobResult)
+
+
+class TestDeployVMsInParallel(cloudstackTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+
+        cls.testClient = super(TestDeployVMsInParallel, cls).getClsTestClient()
+        cls.api_client = cls.testClient.getApiClient()
+
+        cls.testdata = cls.testClient.getParsedTestDataConfig()
+        # Get Zone, Domain and templates
+        cls.domain = get_domain(cls.api_client)
+        cls.zone = get_zone(cls.api_client, cls.testClient.getZoneForTests())
+        cls.hypervisor = cls.testClient.getHypervisorInfo()
+
+        cls._cleanup = []
+
+        cls.template = get_test_template(
+            cls.api_client,
+            cls.zone.id,
+            cls.hypervisor
+        )
+
+        # Create service, disk offerings  etc
+        cls.service_offering = ServiceOffering.create(
+            cls.api_client,
+            cls.testdata["service_offering"]
+        )
+        cls._cleanup.append(cls.service_offering)
+
+        cls.disk_offering = DiskOffering.create(
+            cls.api_client,
+            cls.testdata["disk_offering"]
+        )
+        cls._cleanup.append(cls.disk_offering)
+
+        return
+
+    @classmethod
+    def tearDownClass(cls):
+        super(TestDeployVMsInParallel, cls).tearDownClass()
+
+    def setUp(self):
+
+        self.apiclient = self.testClient.getApiClient()
+
+        self.dbclient = self.testClient.getDbConnection()
+        self.cleanup = []
+        self.hypervisor = self.testClient.getHypervisorInfo()
+        self.testdata["virtual_machine"]["zoneid"] = self.zone.id
+        self.testdata["virtual_machine"]["template"] = self.template.id
+        self.testdata["iso"]["zoneid"] = self.zone.id
+        self.domain = Domain.create(
+            self.apiclient,
+            self.testdata["domain"]
+        )
+        self.cleanup.append(self.domain)
+
+        self.account = Account.create(
+            self.apiclient,
+            self.testdata["account"],
+            domainid=self.domain.id
+        )
+        self.cleanup.append(self.account)
+
+        self.userApiClient = self.testClient.getUserApiClient(UserName=self.account.name, DomainName=self.domain.name)
+        virtual_machine = VirtualMachine.create(
+            self.userApiClient,
+            self.testdata["virtual_machine"],
+            accountid=self.account.name,
+            domainid=self.account.domainid,
+            templateid=self.template.id,
+            serviceofferingid=self.service_offering.id,
+            diskofferingid=self.disk_offering.id,
+            hypervisor=self.hypervisor
+        )
+        self.cleanup.append(virtual_machine)
+        print(f"==== network: {virtual_machine.__dict__}")
```

Comment:
Can we use the `logging` library instead of this? (also in the rest of the print occurrences)

Reply:
yes, good point. I will amend or remove